crates/ide-assists/src/handlers/reorder_fields.rs

```diff
@@ -65,23 +74,31 @@ pub(crate) fn reorder_fields(acc: &mut Assists, ctx: &AssistContext<'_>) -> Opti
         AssistId("reorder_fields", AssistKind::RefactorRewrite),
         "Reorder record fields",
         target,
-        |builder| match fields {
-            Either::Left((sorted, field_list)) => {
-                replace(builder.make_mut(field_list).fields(), sorted)
-            }
-            Either::Right((sorted, field_list)) => {
-                replace(builder.make_mut(field_list).fields(), sorted)
+        |builder| {
+            let mut editor = builder.make_editor(&parent_node);
+
+            match fields {
+                Either::Left((sorted, field_list)) => {
+                    replace(&mut editor, field_list.fields(), sorted)
+                }
+                Either::Right((sorted, field_list)) => {
+                    replace(&mut editor, field_list.fields(), sorted)
+                }
             }
+
+            editor.add_mappings(make.finish_with_mappings());
+            builder.add_file_edits(ctx.file_id(), editor);
         },
     )
 }
 
 fn replace<T: AstNode + PartialEq>(
+    editor: &mut SyntaxEditor,
     fields: impl Iterator<Item = T>,
     sorted_fields: impl IntoIterator<Item = T>,
 ) {
     fields.zip(sorted_fields).for_each(|(field, sorted_field)| {
-        ted::replace(field.syntax(), sorted_field.syntax().clone_for_update())
+        editor.replace(field.syntax(), sorted_field.syntax().clone_for_update())
```

Comment:
We need to eliminate this `clone_for_update` using `SyntaxFactory`. It looks like we will need to add constructors for `record_expr_field` and `record_pat_field`. For now, I would inline this `replace` function so you don't need to worry about how to abstract it over the two cases. We can try to clean it up afterwards.

Reply:
@darichey Generally we'd like to hide the details of creating new nodes under `SyntaxFactory`. In this case though, we're picking nodes from the original syntax tree so we wouldn't need to go through a `SyntaxFactory`.

`SyntaxEditor` should handle transform any node in an edit that reference the original syntax tree into their mutable equivalents, but seems like I forgot this case for the new nodes in replace edits 😅. For now it should be okay to use `clone_for_update` and add a fixme, and then we can remove those fixmes once `SyntaxEditor` is fixed.

Reply:
Got it, thank you! Sorry about that, @tareknaser, you can revert inlining `replace` and just add a FIXME for the clone_for_update :)

Reply:
All set 5c41c20